File: spark-streaming-java-app/src/main/java/com/sapient/learning/MyProperties.java
package com.sapient.learning;

import java.util.Properties;
import java.util.ResourceBundle;

public class MyProperties {
	
	Properties properties = new Properties();
	ResourceBundle rb;
	
	MyProperties() {		
		String env = System.getProperties().getProperty("env");
		
		if (env != null) {
			rb = ResourceBundle.getBundle("application-"+env);
		} else {
			rb = ResourceBundle.getBundle("application");
		}
	}
	
	public String getValue(String key) {
		
		return rb.getString(key);
	}

}

File: spark-streaming-scala-app/src/main/scala/com/sapient/learning/MyProperties.scala
package com.sapient.learning

import com.typesafe.config.ConfigFactory

object MyProperties {
  
  val conf = ConfigFactory.load()
  
  def main(args: Array[String]) {
    
    System.out.println(conf.getString("kafka.server"))
  }
  
  def getValue(key: String): String = {
    
    conf.getString(key)
  }
  
}